app/src/main/java/com/keylesspalace/tusky/db/AccountManager.kt

```diff
@@ -160,6 +170,20 @@ class AccountManager @Inject constructor(db: AppDatabase) {
 
             Log.d(TAG, "updateActiveAccount: saving account with id " + it.id)
             accountDao.insertOrReplace(it)
+
+            // Now that account is known, need to create routing for media-upload failures
+            val dbId = it.id
+            // Remove old observer if we just switched accounts
+            draftsNeedUserAlertCurrent?.let {
+                draftsNeedUserAlert.removeSource(it)
+            }
+            // Add new observer
+            val draftsNeedUserAlertCurrent = draftDao.draftsNeedUserAlert(dbId)
+            this.draftsNeedUserAlertCurrent = draftsNeedUserAlertCurrent
+            draftsNeedUserAlert.addSource(draftsNeedUserAlertCurrent, { count ->
+                Log.d(TAG, "draftsNeedUserAlert: account id " + dbId + " has " + count + " notification-worthy drafts")
+                draftsNeedUserAlert.value = count
```

Comment:
I will remove the debug print before merge. (It's extraneous because this callback is not called if there are no observers.)